tests/functional/text/textPlugin.ts

```diff
@@ -1,49 +1,42 @@
-import * as assert from 'intern/chai!assert';
-import * as registerSuite from 'intern!object';
-import * as Suite from 'intern/lib/Suite';
-import * as Command from 'leadfoot/Command';
+const { registerSuite } = intern.getInterface('object');
+const { assert } = intern.getPlugin('chai');
+import Suite from 'intern/lib/Suite';
 import { Require } from '@dojo/interfaces/loader';
-import pollUntil = require('leadfoot/helpers/pollUntil');
+import pollUntil from '@theintern/leadfoot/helpers/pollUntil';
 
 declare const require: Require;
 
-function executeTest(suite: Suite, htmlTestPath: string, testFn: (result: any) => void, timeout = 10000): Command<any> {
-	return suite.remote
-		.get(require.toUrl(htmlTestPath))
-		.then(pollUntil<any>(function() {
-			return (<any> window).loaderTestResults;
-		}, undefined, timeout), undefined)
-		.then(testFn, function() {
-			throw new Error('loaderTestResult was not set.');
-		});
+async function executeTest(suite: Suite, htmlTestPath: string, timeout = 10000) {
+	try {
+		return await suite.remote.get(htmlTestPath).then(pollUntil(function () {
+			return (<any> window).loaderTestResults || null;
```

Comment:
would it be better to use `dojo/shim/global`?

Reply:
This function is always run on the client and it would be a big overhead to `require()` the shim for this simple polling function.

Reply:
is it a big overhead to require a single module from shim?

Reply:
Since this is a functional tests, the text of this function is sent to the client and run synchronously. There's no way to do a `require(['@dojo/shim/global'], function () { ... })` in the execution on the client for a `pollUntil`.